tests/integration/vlan_test.py

```diff
@@ -64,6 +65,22 @@ def test_add_and_remove_vlan(eth1_up):
     assert not current_state[INTERFACES]
 
 
+@pytest.fixture
+def vlan_with_base_iface(eth1_up):
+    with vlan_interface(VLAN_IFNAME, 101) as desired_state:
+        base_iface_name = desired_state[INTERFACES][0]['vlan']['base-iface']
+        iface_states = statelib.show_only((base_iface_name, VLAN_IFNAME))
+        yield dict(iface_states=iface_states)
```

Comment:
Since it is only one state that is passed, you do not need to use a dictionary here. Then accessing the data in the test function will also be a little bit easier (one dict lookup less).